bigquery_etl/cli/metadata.py

```diff
@@ -126,3 +128,51 @@ def publish(name: str, sql_dir: Optional[str], project_id: Optional[str]) -> Non
             print("No metadata file for: {}.{}.{}".format(project, dataset, table))
 
     return None
+
+
+@metadata.command(
+    help="""
+    Deprecate BigQuery table by updating metadata.yaml file.
+    Deletion date is by default 3 months from current date if not provided.
+
+    Example:
+     ./bqetl metadata deprecate ga_derived.downloads_with_attribution_v2 --deletion_date=2024-03-02
+    """
+)
+@click.argument("name")
+@project_id_option(
+    ConfigLoader.get("default", "project", fallback="moz-fx-data-shared-prod")
+)
+@sql_dir_option
+@click.option(
+    "--deletion_date",
+    "--deletion-date",
+    "-d",
+    help="Date when table is scheduled for deletion. Date format: yyyy-mm-dd",
+    type=click.DateTime(formats=["%Y-%m-%d"]),
+    default=datetime.today() + relativedelta(months=+3),
+)
+def deprecate(
+    name: str,
+    sql_dir: Optional[str],
+    project_id: Optional[str],
+    deletion_date,
```

Comment:
Suggest type hint here for consistency.

Reply:
I'll also add that `sql_dir` and `project_id` are optional as command line args but they're not expected to be null in the function.  I think `paths_matching_name_pattern` would actually error if sql_dir was null.

This might cause false warnings in code inspectors but maybe not in this case since `paths_matching_name_pattern` doesn't use type hints.